packages/webview_flutter/tizen/src/webview.cc

```diff
@@ -555,271 +504,189 @@ static LWE::KeyValue EcoreEventKeyToKeyValue(const char* ecore_key_string,
   }
-
-  LOG_DEBUG("WebViewEFL - unimplemented key %s\n", ecore_key_string);
+  LOG_WARN("Unknown key name: %s", ecore_key.c_str());
   return LWE::KeyValue::UnidentifiedKey;
 }
 
 void WebView::DispatchKeyDownEvent(Ecore_Event_Key* key_event) {
-  std::string key_name = key_event->keyname;
-  LOG_DEBUG("ECORE_EVENT_KEY_DOWN [%s, %d]\n", key_name.c_str(),
-            (key_event->modifiers & 1) || (key_event->modifiers & 2));
-
   if (!IsFocused()) {
-    LOG_DEBUG("ignore keydown because we dont have focus");
     return;
   }
 
-  if ((strcmp(key_name.c_str(), "XF86Exit") == 0) ||
-      (strcmp(key_name.c_str(), "Select") == 0) ||
-      (strcmp(key_name.c_str(), "Cancel") == 0)) {
-    if (strcmp(key_name.c_str(), "Select") == 0) {
-      webview_instance_->AddIdleCallback(
-          [](void* data) {
-            WebView* view = (WebView*)data;
-            LWE::WebContainer* self = view->GetWebViewInstance();
-            LWE::KeyValue kv = LWE::KeyValue::EnterKey;
-            self->DispatchKeyDownEvent(kv);
-            self->DispatchKeyPressEvent(kv);
-            self->DispatchKeyUpEvent(kv);
-            view->HidePanel();
-          },
-          this);
-    } else {
-      webview_instance_->AddIdleCallback(
-          [](void* data) {
-            WebView* view = (WebView*)data;
-            view->HidePanel();
-          },
-          this);
-    }
-  }
+  std::string key_name = key_event->keyname;
+  bool is_shift_pressed = key_event->modifiers & 1;
 
   struct Param {
     LWE::WebContainer* webview_instance;
     LWE::KeyValue key_value;
   };
-  Param* p = new Param();
-  p->webview_instance = webview_instance_;
-  p->key_value =
-      EcoreEventKeyToKeyValue(key_name.c_str(), (key_event->modifiers & 1));
+  Param* param = new Param();
+  param->webview_instance = webview_instance_;
+  param->key_value = KeyNameToKeyValue(key_name, is_shift_pressed);
 
   webview_instance_->AddIdleCallback(
       [](void* data) {
-        Param* p = (Param*)data;
-        p->webview_instance->DispatchKeyDownEvent(p->key_value);
-        p->webview_instance->DispatchKeyPressEvent(p->key_value);
-        delete p;
+        Param* param = reinterpret_cast<Param*>(data);
+        param->webview_instance->DispatchKeyDownEvent(param->key_value);
+        param->webview_instance->DispatchKeyPressEvent(param->key_value);
+        delete param;
       },
-      p);
+      param);
 }
 
 void WebView::DispatchKeyUpEvent(Ecore_Event_Key* key_event) {
-  std::string key_name = key_event->keyname;
-  LOG_DEBUG("ECORE_EVENT_KEY_UP [%s, %d]\n", key_name.c_str(),
-            (key_event->modifiers & 1) || (key_event->modifiers & 2));
-
   if (!IsFocused()) {
-    LOG_DEBUG("ignore keyup because we dont have focus");
     return;
   }
 
+  std::string key_name = key_event->keyname;
+  bool is_shift_pressed = key_event->modifiers & 1;
+
   struct Param {
     LWE::WebContainer* webview_instance;
     LWE::KeyValue key_value;
   };
```

Comment:
I don't see a problem with this part. How should I change this? The type is defined only for use in the local scope on purpose. Sharing the type across different scopes doesn't seem to be a good idea.

Reply:
OK, this plugin improved a lot now by your PR, so I'm satisfied with it even if I leave it as it is.
However, in my personal opinion, since param classes for handling key up/down event are the same, I thought that there should be only one :)